### PR TITLE
lmdb: fix size_t size issues on 32 bit

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/lmdb/value_stream.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ namespace lmdb
     {
         std::size_t count(MDB_cursor* cur)
```
Comment (vtnerd) [inline src/lmdb/value_stream.cpp]: The return type should be updated to `mdb_size_t` or `std::uint64_t`. I think `mdb_size_t` makes most sense, as the header already has the LMDB includes. I'll have to look to call if any callers need to update ...

**Comment by italocoin-project** [comment]
On raspberry there is this error
`database.cpp:139:83: error: no matching function for call to ‘min(mdb_size_t, const size_t&)’
         const mdb_size_t resize = std::min((mdb_size_t)info.me_mapsize, max_resize);`

**Comment by vtnerd** [comment]
The first error is due to the cast being on the wrong parameter. The second you pointed to me (yesterday?), but I wasn't immediately sure why the error is being generated

**Comment by vtnerd** [comment]
Sorry, another comment to mention @moneromooo-monero - the cast is on the wrong parameter in this PR.

**Comment by moneromooo-monero** [comment]
Huh, how did this end up working :S Fixing.

**Comment by moneromooo-monero** [comment]
Ah, that's because I changed the type after the patch. I made both params mdb_size_t, so no cast is needed now. We'll see if the 32 bit build bot is happy with that.

**Comment by italocoin-project** [comment]
Now there is another error

`home/travis/build/monero-project/monero/src/lmdb/error.cpp: In member function 'virtual std::error_condition {anonymous}::category::default_error_condition(int) const':
/home/travis/build/monero-project/monero/src/lmdb/error.cpp:58:39: error: 'state_not_recoverable' is not a member of 'std::errc'
                     return std::errc::state_not_recoverable;`

**Pull request merged**